django/utils/tree.py

```diff
@@ -71,7 +71,7 @@ def __eq__(self, other):
         )
 
     def __hash__(self):
-        return hash((self.__class__, self.connector, self.negated) + tuple([
+        return hash((self.__class__, self.connector, self.negated, *[
```

Comment:
The original `tuple([...])` didn't need the list comprehension and could have been a generator expression.
Is this any better being `*(...)` instead of `*[...]`?

Reply:
It's complicated:
```
In [28]: r = range(10000000)

In [29]: %timeit tuple((x for x in r),)
663 ms ± 933 µs per loop (mean ± std. dev. of 7 runs, 1 loop each)
In [35]: %timeit tuple([x for x in r])
527 ms ± 1.01 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)
In [30]: %timeit (*(x for x in r),)
755 ms ± 1.68 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)
In [31]: %timeit (*[x for x in r],)
609 ms ± 857 µs per loop (mean ± std. dev. of 7 runs, 1 loop each)

In [32]: %memit tuple((x for x in r),)
peak memory: 385.38 MiB, increment: 340.79 MiB
In [36]: %memit tuple([x for x in r])
peak memory: 506.83 MiB, increment: 462.23 MiB
In [33]: %memit (*(x for x in r),)
peak memory: 501.88 MiB, increment: 457.29 MiB
In [34]: %memit (*[x for x in r],)
peak memory: 516.88 MiB, increment: 472.29 MiB
```